server/src/core/user/user.entity.ts

```diff
@@ -63,10 +63,14 @@ export class User {
   deletedAt: Date;
 
   @Field(() => Workspace, { nullable: false })
-  @ManyToOne(() => Workspace, (workspace) => workspace.users)
+  @ManyToOne(() => Workspace, (workspace) => workspace.users, {
+    onDelete: 'SET NULL',
```

Comment:
This part I'm not fully sure, we couldn't delete workspaces because there is a foreign key on the user table with defaultWorkspaceId. We could delete the user in cascade however I'm guessing this defaultWorkspaceId can be changed in the future. However in the meantime that means if someone deletes a workspace and if that person or another workspace-member tries to signIn it will pass the login but will probably fail later due to the missing workspace